src/main/java/javaslang/collection/Iterator.java

```diff
@@ -166,6 +167,16 @@ public T next() {
         };
     }
 
+    @Override
+    default Iterator<T> iterator() {
+        return this;
```

Comment:
The same with ofAll(Iterator) and ofAll(Iterable) they are effectively the identity and will be removed. I created an issue yesterday. Then the method `iterator()` will be removed completely!

You can do this (remove ofAll(Iterable), ofAll(Iterator) and iterator()).

When we need elsewhere an Iterable and we have already an Iterator we just do. `() -> myIterator`